beacon-chain/p2p/service.go

```diff
@@ -406,6 +402,28 @@ func (s *Service) FindPeersWithSubnet(index uint64) (bool, error) {
 	return exists, nil
 }
 
+// Waits for the beacon state to be initialized, important
+// for initializing the p2p service as p2p needs to be aware
+// of genesis information for peering.
+func (s *Service) awaitStateInitialized() {
+	stateChannel := make(chan *feed.Event, 1)
+	stateSub := s.stateNotifier.StateFeed().Subscribe(stateChannel)
+	defer stateSub.Unsubscribe()
+	for {
+		select {
+		case event := <-stateChannel:
+			if event.Type == statefeed.ChainStarted {
```

Comment:
need to handle also when the chain is already started, so `statefeed.Initialized`